### ISSUE TITLE
Error not detected in subshell

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1718]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2012
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

# This one properly complains about SC2012
ls -la "$@"| zenity --text-info --auto-scroll --width=1320 --height=800 --title "ls -la $*" &

# This one does not complain about SC2012.
(ls -la "$@") | zenity --text-info --auto-scroll --width=1320 --height=800 --title "ls -la $*" &

```

#### Here's what shellcheck currently says:
```
Line 4	SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```
#### Here's what I wanted or expected to see:

```
Line 4	SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
Line 7	SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```
The second example should complain about SC2012; there should be no difference if it's run in a subshell, no?
